docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+      "platform": { 
+        "architecture": "amd64", 
+        "os": "linux" 
+      } 
+    }, 
+    { 
+      "mediaType": "application/vnd.oci.image.manifest.v1+json", 
+      "digest": "sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a", 
+      "size": 1239, 
+      "platform": { 
+        "architecture": "arm64", 
+        "os": "linux" 
+      } 
+    } 
+  ] 
+} 
+```
+
+4. Add (or change) annotations to an image index in the OCI layout: 
+
+```bash
+$ oras manifest index update --oci-layout layout-dir:v1 --annotation “platform=multi-arch” 
+
+Resolving  v1 
+Resolved   sha256:2af402374d4c9297bf077b1e722d523c5ce37256d1d04 
+Updating anootation  v1 
+Updated    sha256:447c006400aae8b0d302cafba172f5da9889d50af085a 
+Pushed     [oci-layout] layout-dir:v1 
+Digest: sha256:447c006400aae8b0d302cafba172f5da9889d50af085a 
+Updated annotations of layout-dir:v1 
+```
+
+5. Attach an annotation as a referrer to an existing multi-arch image and its child arch-specific image manifest (without mutating its digest) in one shot. The same annotation is supposed to be added to the parent image index and its child arch-specific image manifest recursively: 
+
+
+```bash
+oras attach --artifact-type application/vnd.artifact.lifecycle  --annotation "vnd.artifact.lifecycle.end-of-life.date=2023-05-12" --oci-layout layout-dir:v1  
+
+Attached to [oci-layout] layout-dir@sha256:2af402374d4c9297bf077b1e722d52 
+Digest: sha256:117308d626166e77ffbd9c76b5545101b723csdcxcxc2344556642 
+```
+
+6. View attached annotations of the multi-arch image (index) and its child image manifest: 
+ 
+```bash
+$ oras discover --oci-layout layout-dir:v1 --recursive --format tree-full  
+
+Image index
+layout-dir@sha256:447c006400aae8b0d302cafba172f5da9889d50af085ad51d 
+└── application/vnd.artifact.lifecycle 
+    └── sha256:b25b53b7df86a5dc7087cf96ed88fd1d8872271422224d7 
+        └── vnd.artifact.lifecycle.end-of-life.date: "2023-05-12
+
+Image manifest [1] 
+layout-dir@sha256:447c006400aae8b0d302caf9d50af089d50af089d50af089d 
+└── application/vnd.artifact.lifecycle 
+    └── sha256:b25b53b7df86a5dc7087cf96ed88fd1d8872271422224d7 
+        └── vnd.artifact.lifecycle.end-of-life.date: "2023-05-12" 
+
+ 
+
+Image manifest [2] 
+layout-dir@sha256:447c006400aae8b0d302cafba172f5da9889d50af085ad51d 
+└── application/vnd.artifact.lifecycle 
+    └── sha256:b25b53b7df86a5dc7087cf96ed88fd1d8872271422224d7 
+        └── vnd.artifact.lifecycle.end-of-life.date: "2023-05-12" 
+
+```
+
+![annotation as referrers](./img/annotation-as-referrers.png)
+
+1. List the tags in the OCI image layout. The digest and platform information should be listed alongside the tags. 
+
+```bash
+$ oras repo tags --oci-layout layout-dir 
+```
+
+
+| Tag  | Digest  | Platform   |
+|--- | --- |  ---  |
+| v1 | sha:256@xxxx | multi-arch (linux/amd64, linux/arm64)  |
+| v1-linux-amd64  |    sha:256@yyyy          |     linux/amd64   |
+| v1-linux-arm64  |    sha:256@zzzz          |     linux/arm64   |
```

Comment:
Is this the command output or just a table describe the internals? It is confusing from the doc point of view.

Reply:
It is output. I called it out in this paragraph